### ISSUE TITLE
Fails to extract header row when FixedWidthParser with annotations 

### ISSUE THREAD

**Issue by Fadelis** [uniVocity/univocity-parsers#180]
When trying to parse a fixed width file with `BeanListProcessor` using annotations and having enabled `headerExtraction` it still tries to map header row to a bean.
If the length and headers are defined through `FixedWidthParserSettings` the header row is extracted.

```
  public static void main(final String... args) {
    final String testValue = new StringBuilder()
        .append("Name      Surname        Age\n")
        .append("John      Smith          25 \n")
        .append("Richard   Corrington     25 \n")
        .toString();

    final BeanListProcessor<TestDTO> rowProcessor = new BeanListProcessor<>(TestDTO.class);
    final FixedWidthParserSettings parserSettings = new FixedWidthParserSettings();
    parserSettings.getFormat().setPadding(' ');
    parserSettings.setProcessor(rowProcessor);
    parserSettings.setHeaderExtractionEnabled(true);
    //create parser based on settings type
    new FixedWidthParser(parserSettings).parse(new StringReader(testValue));

    final List<TestDTO> records = rowProcessor.getBeans();
  }

  public static class TestDTO {
    @Parsed(index = 0)
    @FixedWidth(value = 10)
    private String name;
    @Parsed(index = 1)
    @FixedWidth(value = 15)
    private String surname;
    @Parsed(index = 2)
    @FixedWidth(value = 3)
    private Integer age;

    public TestDTO() {
    }
  }
```

**Comment by Fadelis** [comment]
This was found with version `2.4.1`. Seems to be working on version `2.5.2`.

**Comment by jbax** [comment]
Thank you for reporting this issue. Fixed and released a 2.5.3-SNAPSHOT build to include the adjustment.
